add #[openapi(rename = "...")] attribute

tomsik68 opened the pull request:
Hello! 

I designed a fix for #357 which is a little different from the approach in #358. This time, I'm adding the explicit rename using `#[openapi(rename = "...")]`. If there's a good way to test my approach using existing paperclip tests, I'd love to know about it.

tiagolobocastro commented on macros/src/actix.rs:
```diff
@@ -659,6 +659,44 @@ pub fn emit_v2_errors_overlay(attrs: TokenStream, input: TokenStream) -> TokenSt
     gen.into()
 }
 
+fn extract_rename(attrs: &[Attribute]) -> Option<String> {
```
We already have an `extract_openapi_attrs` - could we make use of it here?
Btw, while you're at it could you replace the string "openapi" in `extract_openapi_attrs` with `SCHEMA_MACRO_ATTR`?

tiagolobocastro commented on macros/src/actix.rs:
```diff
@@ -659,6 +659,44 @@ pub fn emit_v2_errors_overlay(attrs: TokenStream, input: TokenStream) -> TokenSt
+                    {
+                        match mnv.lit {
+                            Lit::Str(s) => Some(s.value().clone()),
+                            _ => panic!("#[openapi(rename = ...)] expects literal as argument"),
```
Should probably use `emit_error` instead of panic

tiagolobocastro commented on macros/src/actix.rs:
```diff
@@ -659,6 +659,44 @@ pub fn emit_v2_errors_overlay(attrs: TokenStream, input: TokenStream) -> TokenSt
+                .map(|id| id.to_string() == "openapi")
+                .unwrap_or(false)
+        })
+        .next()
```
This doesn't seem right, we should probably search the it as it might not always the first one? (we could add other things in the future).
Maybe have a look at `emit_v2_security` which uses the openapi attribute already.

tiagolobocastro commented:
> Hello!
> 
> I designed a fix for #357 which is a little different from the approach in #358. This time, I'm adding the explicit rename using `#[openapi(rename = "...")]`. If there's a good way to test my approach using existing paperclip tests, I'd love to know about it.

So the tests basically take your "config", generate the openapi spec and compare it against a hardcoded spec using `check_json`. I see you've renamed a Pet - that should fail the tests now are they expect to see `Pet` and not `PetRenamed` - however please add a new test instead to avoid changing all the existing tests which expect `Pet`.

tomsik68 commented:
Hey @tiagolobocastro, I rebased, reimplemented `extract_rename` using your suggestions and added a test case for it! Could you please take another look when you have a chance? Thanks!

tiagolobocastro commented:
awesome thanks @tomsik68, I've kicked gh actions and will try to have a look during the week

tiagolobocastro commented:
hey @tomsik68 would you mind fixing the clippy errors [here](https://github.com/paperclip-rs/paperclip/runs/4929089935?check_suite_focus=true)

tiagolobocastro commented:
Thank you @tomsik68 !
